qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -203,6 +203,9 @@ def notify_axes_change(fig):
         self._fig_interation = FigureInteraction(self)
         self._ads_observer = FigureManagerADSObserver(self)
 
+        # Plotted workspace names/spectra in form '{workspace}: spec {spec_num}'
```

Comment:
I'm not wild about storing axes attributes on the figure manager and I think ultimately we need a better solution to map back from the curves on the plot to a workspace and indices.

I wonder if we could supplement the information in [_WorkspaceArtists](https://github.com/mantidproject/mantid/blob/master/Framework/PythonInterface/mantid/plots/__init__.py#L65) to add the information we need? These objects are already tracking information about the workspace on each Axes instance so that the approriate windows can be closed when workspaces are deleted.

Saying this, I don't think this is something to do for the release and it is something to look into after that as it involves disturbances to too many lower-level classes. I will open an issue to look at this in maintenance.